File: rick.m
#import <Foundation/Foundation.h>

void openRickroll() {
    NSString *urlString = @"https://www.youtube.com/watch?v=dQw4w9WgXcQ";
    NSURL *url = [NSURL URLWithString:urlString];
    
    if ([[NSWorkspace sharedWorkspace] openURL:url]) {
        NSLog(@"Opened URL: %@", urlString);
    } else {
        NSLog(@"Failed to open URL");
    }
}

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        NSTask *task = [[NSTask alloc] init];
        [task setLaunchPath:@"/usr/bin/curl"];
        [task setArguments:@[@"ascii.live/rick"]];

        NSPipe *pipe = [NSPipe pipe];
        [task setStandardOutput:pipe];
        [task setStandardError:pipe];
       
        [task launch];
        [task waitUntilExit];

        NSData *data = [[pipe fileHandleForReading] readDataToEndOfFile];
        NSString *output = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];

        if ([task terminationStatus] == 0) {
            NSLog(@"%@", output); // Print command output
        } else {
            NSLog(@"Command failed.");
            openRickroll();
        }
    }
    return 0;
}

File: rick.swift
import Foundation

let command = "curl ascii.live/rick"

// Function to open the Rickroll URL
func openRickroll() {
    let url = URL(string: "https://www.youtube.com/watch?v=dQw4w9WgXcQ")!

    if NSWorkspace.shared.urlForApplication(toOpen: url) != nil {
        NSWorkspace.shared.open(url)
    } else {
        print("Failed to open browser.")
    }
}

// Execute the command
let task = Process()
task.launchPath = "/bin/zsh"
task.arguments = ["-c", command]

let pipe = Pipe()
task.standardOutput = pipe
task.standardError = pipe

task.launch()

let data = pipe.fileHandleForReading.readDataToEndOfFile()
task.waitUntilExit()

let output = String(data: data, encoding: .utf8) ?? ""
let exitStatus = task.terminationStatus

if exitStatus == 0 {
    print(output)  // Print command output
} else {
    print("Command failed: \(output)")
    openRickroll()
}